src/Lecture3.hs

```diff
@@ -125,10 +137,10 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) (Reward g1 special1) (Reward g2 special2) = Reward (g1 <> g2) (special1 || special2) 
 
 instance Monoid Reward where
-
+  mempty = Reward (mempty :: Gold) False
```

Comment:
You don't need to specify the `Gold` type here explicitly 🙂  Haskell has type inference so it's able to infer the correct type of `mempty`
```suggestion
  mempty = Reward mempty False
```

Reply:
Yeah I did think about that but I thought that maybe specifying the type makes it more readable ?!? Whats the usual convention ? @chshersh

Reply:
Sometimes specifying the type signature explicitly indeed helps to understand code better in the future and is recommended. In the `Monoid` instances, however, it generally doesn't matter because the default expected implementation for a data type with a multiple-fields constructor is just `mempty` for all fields.